polaris/viz/planar.py

```diff
@@ -101,6 +125,19 @@ def plot_horiz_field(ds, ds_mesh, field_name, out_file_name=None,  # noqa: C901
     patch_mask : numpy.ndarray
         A mask used to select entries in the field that have patches
     """
+    if field_name not in ds:
+        raise ValueError(
+            f'{field_name} must be present in ds before plotting.')
+
+    if patches is not None:
+        if patch_mask is None:
+            raise ValueError('You must supply both patches and patch_mask '
+                             'from a previous call to plot_horiz_field()')
+
+    if (transect_x is None) != (transect_y is None):
+        raise ValueError('You must supply both transect_x and transect_y or '
+                         'neither')
+
```

Comment:
I just consolidated these at the top because it's a good practice to raise errors early before wasting any time.